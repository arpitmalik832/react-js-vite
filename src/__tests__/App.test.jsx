import { fireEvent, render } from '@testing-library/react';
import App from '../App';

describe('App', () => {
  it('snapshot test for component', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });

  it('renders the app component', () => {
    const { getByTestId } = render(<App />);

    fireEvent.click(getByTestId('button'));
  });
});
