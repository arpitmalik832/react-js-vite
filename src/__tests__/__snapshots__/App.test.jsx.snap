// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test for component 1`] = `
<div>
  <div>
    <a
      href="https://vitejs.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="test-file-stub"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="test-file-stub"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <button
      data-cy="button"
      data-testid="button"
    >
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.jsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="readTheDocs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;
